### PR TITLE
Remove stalled (cancelled) trains from extract_stop_dict

### LINKED ISSUE #29: Cancelled trips listed
Love the library, I use it to display some train times on a [small LED board](https://github.com/WardBrian/mta-board) in my apartment. 

I recently noticed it was listing some train times which were not actually present when I walked down to the Station. Apple Maps has those same trains listed, but marked as "Cancelled"

Is there any way to remove cancelled trains from my listing? It seems it must be in the API if Apple has it, but I can't find how

### PR DESCRIPTION
**Pull request by WardBrian**
Closes #29.

Using the same data [I posted there](https://github.com/nolanbconaway/underground/issues/29#issuecomment-1794009909) (I just overwrote `request_robust` to return the same dict I had saved that night), which had some known cancelled trains, they are now omitted.

Before:

```python
>>> underground.SubwayFeed.get('F', api_key=api).extract_stop_dict()['F']['D17S']
[datetime.datetime(2023, 11, 5, 21, 54, 18, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 22, 5, 1, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>), # CANCELLED
 datetime.datetime(2023, 11, 5, 22, 15, 11, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>), 
 datetime.datetime(2023, 11, 5, 22, 32, 1, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>), 
 datetime.datetime(2023, 11, 5, 22, 57, 30, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 23, 12, 30, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 23, 27, 30, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>)]
```
After:
```python
>>> underground.SubwayFeed.get('F').extract_stop_dict()['F']['D17S']              
[datetime.datetime(2023, 11, 5, 21, 54, 18, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 22, 15, 11, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 22, 32, 13, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 22, 57, 30, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 23, 12, 30, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>),
 datetime.datetime(2023, 11, 5, 23, 27, 30, tzinfo=<DstTzInfo 'US/Eastern' EST-1 day, 19:00:00 STD>)]
```

### CODE REVIEW

**Comment by WardBrian** [comment]
Open to suggestions on new tests for this @nolanbconaway

**Review by nolanbconaway — COMMENTED** [review]
changes look fine! one note on the actual handler you wrote, but tests seem sufficient. 

i worry a little bit about false positives from that 90s timeout; tho if the MTA says so...

Some minor annoyances:

- you need to bump the [version](https://github.com/nolanbconaway/underground/blob/master/src/underground/version) file so that this releases to pypi correctly.
- The gh action [busted](https://github.com/nolanbconaway/underground/actions/runs/6917853658/job/18834254845?pr=30#step:4:10).  seems like ubuntu latest switched from 20.04 to 22.04 since the ci last ran, and py3.6 is not available in 22.04. I don't want your changes to get hung up on the py3.6 issue, so the easiest fix is to run the [CI](https://github.com/nolanbconaway/underground/blob/master/.github/workflows/push.yml) on `ubuntu-20.04`. Might as well make that change in the [release to pypi action](https://github.com/nolanbconaway/underground/blob/master/.github/workflows/release_to_pypi.yml) as well.

**Comment by nolanbconaway** [comment]
>I'd be happy to have the timeout be greater (180s?), since I agree that also seems short for a stall. I think it would only really matter for people monitoring the stops near the ends of lines. If a train is marked as stalled in one update and not in the next, chances are it wasn't one of the upcoming ~5 trains for most stops besides the terminals anyway

I think this opens up the idea of a function arg / envvar to set the timeout length (with negative values disabling?). I considered this earlier but did not bring it up because i didn't want to scope creep your pr. I think what you have is fine since thats what the MTA says, but totally welcome you to extend your changes so that this duration is configurable.

> What version would you like me to bump to? 0.3.4?

SGTM

> edit: my reading of this paragraph in the docs is that there is always either 0 or 1 vehicle updates:

Yep SGTM. no need to add a groupby.

> Since 3.6 is past it's end-of-life date would it be better to just update the matrix so it is testing on more recent versions (e.g. 3.11 and 3.12 as well?)

Yeah i was thinking that too, but again didn't want you to have to fix all these CI issues just to merge this. I think your last change is fine for now; but again totally welcome the help if you want to update the packaging and CI.

**Comment by WardBrian** [comment]
I like the configurable timeout. I made the default 90 still, if you'd prefer it to be 0 to keep the existing behavior we could also do that

File: src/underground/models.py [hunk 1]
```diff
@@ -211,35 +211,47 @@ def get(route_or_url: str, retries: int = 100, api_key: str = None) -> "SubwayFe
             )
         )
 
-    def extract_stop_dict(self, timezone: str = metadata.DEFAULT_TIMEZONE) -> dict:
+    def extract_stop_dict(
```
Comment (nolanbconaway) [inline src/underground/models.py:214]: this is great!  do you think this arg should be exposed to the cli as well? by adding an arg here https://github.com/nolanbconaway/underground/blob/master/src/underground/cli/stops.py#L48

Comment (WardBrian) [reply src/underground/models.py:214]: Done!

**Review by nolanbconaway — APPROVED** [review]
LGTM! thanks for putting in the work @WardBrian

**Pull request merged**